app/controllers/custom-forms/instances/index.js

```diff
@@ -59,6 +67,50 @@ export default class CustomFormsInstancesIndexController extends Controller {
     if (!ttl) {
       return;
     }
-    throw new Error('Niet bewaren aub', ttl); // TODO: remove this. Done as ember sees this as a getter
+    throw new Error('Niet bewaren aub', ttl);
+  }
+
+  @action
+  openDeleteFormModal(formModel) {
+    this.isDeleteModalOpen = true;
+    this.formToDelete = formModel;
+  }
+
+  @action
+  async deleteForm() {
+    this.isDeleting = true;
+    try {
+      await this.customForms.removeFormDefinitionWithUsage(this.model.form);
+      showSuccessToast(
+        this.toaster,
+        'Formulier definitie and instances succesvol verwijderd',
+        'Formulier'
+      );
+      this.router.transitionTo('custom-forms.overview');
+    } catch (error) {
+      showErrorToast(
+        this.toaster,
+        'Er liep iets mis bij het verwijderen van het formulier',
+        'Formulier'
+      );
+    }
+    this.isDeleting = false;
+    this.isDeleteModalOpen = false;
+    this.formToDelete = null;
+  }
+
+  get removeFormModalText() {
+    if (this.model.usage.hasUsage) {
+      const usageCount = this.model.usage.usageUris.length;
```

Comment:
would just return the count, else this list can be very long as it's not paginated